test/CMakeLists.txt

```diff
@@ -99,6 +119,18 @@ macro(_add_test _NAME _DEPENDS)
     add_dependencies(${_NAME} ${_NAME}_data)
     add_test(NAME ${_NAME} COMMAND ${_NAME})
     target_link_libraries(${_NAME} fdeep Threads::Threads doctest::doctest)
+    # GNU in linux doesn't support "-mbig-obj" option.
+    if(CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
`CMAKE_COMPILER_IS_GNUCXX` is [deprecated](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCXX.html) in favor of `CMAKE_CXX_COMPILER_ID`.

Reply:
Also, could you please explain to me, why setting additional compiler flags for GNUCXX helps MinGW?

If I understand correctly, for MinGW, this change does not introduce any difference. :monocle_face:

Reply:
yeah I read in an artical [https://blog.csdn.net/10km/article/details/121289182](url) and [https://stackoverflow.com/questions/16596876/gcc-equivalent-of-mss-bigobj](url) but I haven't test in linux. I will check whether GNU in linux can use flag `-mbig-obj`.
Also, I just noticed `CMAKE_COMPILER_IS_GNUCXX` in the url above is deprecated too. I will use `CMAKE_CXX_COMPILER_ID` and test in linux and give another commit.

Reply:
yeah I read in an artical [https://blog.csdn.net/10km/article/details/121289182](url) and [https://stackoverflow.com/questions/16596876/gcc-equivalent-of-mss-bigobj](url) but I haven't test in linux. I will check whether GNU in linux can use flag `-mbig-obj`.
Also, I just noticed `CMAKE_COMPILER_IS_GNUCXX` in the url above is deprecated too. I will use `CMAKE_CXX_COMPILER_ID` and test in linux and give another commit.

Reply:
Thanks, but I still don't quite understand.

In [the issue](https://github.com/Dobiasd/frugally-deep/issues/367) you write:

> Mingw will report Fatal error: can't write 286 bytes to section. text when compiling unittest. It is recommended to add:
> target_ compile_ options(PROJECT_NAME
> PRIVATE
> $&lt;$&lt;CXX_ COMPILER_ ID:MSVC>:/bigobj>
> $&lt;$&lt;CXX_ COMPILER_ ID:GNU>:-Wa,-mbig-obj>)

but here in the pull request, you modify the compiler flags for all GNUCXX compilers, which the majority of are not MinGW.

So in order to make MinGW work, we pass superfluous flags for everybody working with a "normal" Linux?

Reply:
I just test that, on Windows Mingw, GNU support `-Wa,-mbig-obj` flags, `check_cxx_compiler_flag("-Wa,-mbig-obj" GNU_BIG_OBJ_FLAG_ENABLE)` will set `1` when GNU supports `-Wa,-mbig-obj`. This work fine on Windows Mingw (GNU), but on Linux (Centos 9), `GNU_BIG_OBJ_FLAG_ENABLE` is empty and `-Wa,-mbig-obj` won't be added in compile flags.
I have anothe issue that when I compile test in Centos 9 platform, `make` gave me an error `In file included from /home/frugally-deep/test/test_model_variable_test.cpp:8:
/usr/local/include/doctest/doctest.h:4011:47: error: size of array 'altStackMem' is not an integral constant-expression
 4011 |         static char             altStackMem[4 * SIGSTKSZ];
      |                                               ^
make[2]: *** [test/CMakeFiles/test_model_variable_test.dir/build.make:76: test/CMakeFiles/test_model_variable_test.dir/test_model_variable_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:153: test/CMakeFiles/test_model_variable_test.dir/all] Error 2
make: *** [Makefile:146: all] Error 2`
It might be a bug in doctest and conflicts with Centos ?
Because of this I can't test if we don't add `-Wa,-mbig-obj`, the compile  will raise `bigobj error` or not.
So next we just make sure that if we do not add flag `-Wa,-mbig-obj` in "normal" linux, the error of bigobj will raise or not. If not, we just consider MinGW and MSVC on Windows.

Reply:
Thanks for testing all this. :+1:

In the end, frugally-deep is just a header-only library, i.e., you can include it in your project without a build step. So why do you need to use CMake on it with MinGW and invoke the tests, etc.?

Reply:
Oh, and are you going to open an issue in the `doctest` repo about the `altStackMem` error?

Reply:
HAHAHA, yeah.I have a hobby that when I see a CMakeLists Txt, I just wants to use it and make various modifications. However, the first time a bug is reported by Big obj, I have not seen it before, and I am very interested. I think that many people may encounter this problem, so I want to mention it. During this period, I have had a very pleasant communication with you. Your preciseness so impressed me. I will open an issue in the future about the doctest error.

Reply:
Cool hobby. :grin: :+1:

But I have a hobby too, and this is:

Keeping the complexity of the projects, which I maintain, as low as possible. :wink:

Reply:
I feel, this if block (`if(CMAKE_COMPILER_IS_GNUCXX) ...` is the main thing blocking this MR from being merged, and it's not going to make it. If you remove it (and fix the minor phrasing things), we're good to go. :+1: